### ISSUE TITLE
[Feature Request] Offer other weather providers like Yr

### ISSUE THREAD

**Issue by fdw** [WangDaYeeeeee/GeometricWeather#242]
Thanks for the app, it looks really nice 🙂
However, OpenWeatherMap and Météo France never load data for me and Accuweather is almost ridiculously wrong. How much effort would it be to add, for example, [Yr](https://www.yr.no/en) ([developer portal](https://developer.yr.no/))?

**Comment by papjul** [comment]
Météo France won't be very useful outside France.

Yr can be useful if free of charge. I can have a look at it after current PRs are merged, no guarantee though, it's time consuming.

I recommand that you open separate issues for other weather providers you had in mind ;)

**Comment by fdw** [comment]
As far as I can see, Yr is free of charge. They ask for attribution and an identifying user agent, and one must not make too many requests. I don't know how popular GeometricWeather is or how many requests it makes, so I can't guess whether that would be a problem. (Source: https://developer.yr.no/doc/TermsOfService/)

Unfortunately, I'm not really experienced with Android, otherwise I might try myself.

**Comment by papjul** [comment]
User-Agent is not an issue but I can see many requirements that would require development:

* Do not ask too often, and don't repeat requests until the time indicated in the `Expires` response header. > this requires to forbid manual refresh by the user until the date indicated in the `Expires` response.
* Cache data locally and and use the `If-Modified-Since` request header to avoid repeatedly downloading the same data. > Data is cached locally, but I'm almost sure it doesn't make use of `If-Modified-Since`
* Don't generate unnecessary traffic, e.g. by repeated requests for data which never change (MetAlerts CAP files, images with timestamps and so on). > I don't think we use anything like that
* Don't schedule many requests at the same time, e.g. every hour on the dot or when the forecast model runs are finished. Add a random number of minutes to the time of the requests as our data are continously updated. Spread your traffic evenly out over time so it makes a flat curve, not a sawtooth. > This doesn't apply to us, the automatic refresh is scheduled every X hours from the initial refresh. For example, initial call at 13:04 means next call at 15:04.
* When using requests with latitude/longitude, truncate all coordinates to max 4 decimals. There is no need to ask for weather forecasts with nanometer precision! For new products, requests with 5+ decimals will return a 403 Forbidden. > This requires a small development
* Avoid continuous updating of mobile devices. Applications on mobile devices must not retrieve new data as long as the application is not in use. If you need push notification (e.g. for MetAlerts), don't make more than one poll every 10 mins. > This is probably the hardest one. Geometric Weather was conceived in a way that makes background usage. I would suggest to open a specific issue to request that if app is not in use, automatic refresh only applies to weather alerts refresh.

**Comment by legacychimera247** [comment]
What about using foreca?it's generally very accurate...

**Comment by papjul** [comment]
> What about using foreca?it's generally very accurate...

Free account is for up to 2000 calls per day, which is ridiculously low when you think that 1 000 000 calls per month was reached in a few days on Open Weather Map. If you still want this provider, please open a new issue.

**Comment by legacychimera247** [comment]
Nah, i mean, it would be a good replacement for openweathermap which is not working but i guess is too much of a trouble

It's a shame accuweather is always wrong in my area...
